join-statements/joinstatements.md

```diff
@@ -0,0 +1,423 @@
+
+```
+SELECT students.name societies.society
+FROM students
+LEFT JOIN societies ON students.id = societies.student_id;
+```
+
+**RIGHT OUTER JOIN** - similar to LEFT OUTER JOIN, except instead of focusing on the table specified in the `FROM` statement (in the previous case it was table A), we would focus on the table linked to the  `RIGHT OUTER JOIN` statement, in this case, table B. Similarly, there will be a single row in the output for every row in table B, and any row that does not meet the conditional in the `ON` statement will have null values in the output values of columns from table A.
+
+In the extracurricular society example shown before, using a RIGHT OUTER JOIN would yield an output similar to the INNER JOIN output, _plus_ if we've got societies with no members, they would show up. This would be useful for example to pinpoint societies for students interested in building a society from the ground up.
+
+```
+SELECT students.name societies.society
+FROM students
+RIGHT JOIN societies ON students.id = societies.student_id;
+```
+
+**FULL OUTER JOIN** - It's the combination of an INNER JOIN, LEFT OUTER JOIN, and RIGHT OUTER JOIN.
+
+Similar to the LEFT OUTER JOIN, RIGHT OUTER JOIN, and the INNER JOIN. We would get the output of the LEFT JOIN, and the RIGHT JOIN and would be useful in being able to easily find the students that have no society, and perhaps also point out to them which societies they would be able to make the most impact in by helping create it from the ground up.
+
+```
+SELECT students.name societies.society
+FROM students
+FULL OUTER JOIN societies ON students.id = societies.student_id;
+```
+
+3) **Primary key** would be a key that is a unique identifier in a table, and is only found once within the entire table. This would allow you to identify a single row of data just by it's **primary key**.
+
+A **primary key** real world example would be in a college, your student ID #. This would be your primary key in the College's `students` database table.
+
+A **foreign key** would be a key in a table that links entries from this table to another table via the other table's *primary key*. Typically used in many to one relationships where we could link multiple of the same *primary keys* in a table by assigning it to a *foreign key* value.
+
+A **foreign key** real world example would be your major's program ID found within your Colleges *students* table. Alternatively, the foreign key could also be a society that the student belongs to. The major ID or society ID would be the *primary key* in the College's `program major` table or the `societies` table respectively. The major ID or society ID could both be **foreign keys** in the `students` table.
+
+4) Aliasing is utilizing a different variable name to represent a table's name or a column's name. Typically done by using the statement `AS` when declaring a `FROM` or `JOIN` statement for  representing a table name, such as:
```

Comment:
Why is a different variable better than using the table's name?

Reply:
done